BingAds/shopping-content/code-example-manage-products.md

```diff
@@ -1754,4 +1755,180 @@ def random_string(length=6):
+    $result = request('GET', $url);
+    if ($result === FALSE) {  
+        throw new Exception(var_dump($result));
+    } else {
+        $catalogs = json_decode($result, TRUE)["catalogs"];
+        return $catalogs;
+    }    
+}
+
+function list_products($nextPageToken = null){
+    global $merchantId, $productsUri;
+    # max-results set to 2 to test paging, this would be set to a
+    # higher value in a real world scenario based on your needs.
+    $productsQueryString = '?max-results=2&alt=json';
+    $productsStartTokenQueryString = $productsQueryString . '&start-token=%s';
+
+    $url = sprintf($productsUri . $productsQueryString, $merchantId);
+    $results = array();
+    while ($url !== null) {
+        $result = request('GET', $url);
+        if ($result === FALSE) {  
+            throw new Exception(var_dump($result));
+        }
+        $productsResponse = json_decode($result, TRUE);
+        
+        foreach ($productsResponse['resources'] as $key => $value) {
+            array_push($results, $value);
+        }
+        if ($productsResponse !== null && array_key_exists('nextPageToken', $productsResponse)) {
+            $url = sprintf($productsUri . $productsStartTokenQueryString, $merchantId, $productsResponse['nextPageToken']);
+        } else {
+            $url = null;
+        }
+    }
+    return $results;
+}
+
+function add_product($catalogId, $product){
+    global $merchantId, $productsUri;
+    
+    $url = sprintf($productsUri . '?bmc-catalog-id=%s&alt=json', $merchantId, $catalogId);
+    $result = request('POST', $url, $product);
+    if ($result === FALSE) {  
+        throw new Exception(var_dump($result));
+    } 
+    return json_decode($result, TRUE);    
+}
+
+function get_product($productId){    
+    global $merchantId, $productUri;
+    $url = sprintf($productUri . "?alt=json", $merchantId, $productId);
+    $result = request('GET', $url);
+    if ($result === FALSE) {  
+        throw new Exception(var_dump($result));
+    } 
+    return json_decode($result, TRUE); 
+}
+
+function delete_product($productId){
+    global $merchantId, $productUri;
+    $url = sprintf($productUri . "?alt=json", $merchantId, $productId);
+    $result = request('DELETE', $url);
+    if ($result === FALSE) {  
+        throw new Exception(var_dump($result));
+    } 
+}
+
+function request($method, $url, $data = null){
+    global $devToken, $authenticationToken;
+    $options = array(
+        'http' => array(
+            'method'  => $method,
+            'header'  => "AuthenticationToken: $authenticationToken\r\n" .
+                        "DeveloperToken: $devToken\r\n" .
+                        "Content-type: application/json\r\n"
+        )
+    );
+    $json = null;
+    if($data !== null){
+        $json = json_encode($data);
+        $options['http']['content'] = $json;
+    }
+    $context  = stream_context_create($options);
+    echo("requesting ($method): $url\r\n");
+    if($json !== null){
```

Comment:
just noticed this one also, I have another php example coming and I'll update this spacing also